helix-term/src/ui/editor.rs

```diff
@@ -1086,8 +1083,9 @@ impl Component for EditorView {
             );
         }
 
-        if let Some(ref mut info) = self.autoinfo {
+        if let Some(mut info) = cx.editor.autoinfo.take() {
             info.render(area, surface, cx);
+            cx.editor.autoinfo = Some(info)
         }
```

Comment:
Why the need for `take()` here?

Reply:
Ah I see, it sits on the editor now

Reply:
It could be an extra field on the context? That way UI is fully stored on the `ui::Editor`